io-engine/src/bdev/nexus/nexus_bdev.rs

```diff
@@ -1336,6 +1391,60 @@ pub async fn nexus_create_v2(
     }
 }
 
+async fn get_nexus_zone_info_from_children(children_devices: &mut Vec<(String, String)>) -> Result<ZoneInfo, Error> {
+    let mut zone_info = ZoneInfo::default();
+    let mut conventional = false;
+    let mut info_set = false;
+
+    for (_uri, device_name) in &*children_devices {
+        let dev = device_lookup(&device_name).unwrap();
+
+        if dev.is_zoned() {
+            zone_info.zoned = true;
+            if !info_set {
+                zone_info.num_zones = dev.get_num_zones();
```

Comment:
It looks like adding a couple of helper methods to copy and compare zone info would make code much cleaner.